### PR TITLE
[PM-7809] Fix memory leak in AngularThemingService for Safari extension

### PR DESCRIPTION
**Pull request by dani-garcia**
## 🎟️ Tracking
https://bitwarden.atlassian.net/browse/PM-7809

## 📔 Objective

Fixes the memory leak in the Safari extension caused by holding a reference to the background page in AngularThemingService.

To avoid holding the reference we have to disable the system theme observable, but that never worked in Safari anyway.

There was another memory leak caused by the use of `getBgService` for AuthService, but that got fixed last week in https://github.com/bitwarden/clients/pull/9315

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/angular/src/platform/services/theming/angular-theming.service.ts [hunk 1]
```diff
@@ -31,6 +31,17 @@ export class AngularThemingService implements AbstractThemingService {
...
+  /**
+   * Gets the currently active system theme based on the given window.
+   * @param window The window to query for the current theme.
+   * @returns The active system theme.
+   */
+  static getSystemThemeFromWindow(window: Window): ThemeType {
+    return window.matchMedia("(prefers-color-scheme: dark)").matches
+      ? ThemeType.Dark
+      : ThemeType.Light;
+  }
```
Comment (justindbaur) [inline libs/angular/src/platform/services/theming/angular-theming.service.ts:43]: Can you call this function inside the `of` in the above `createSystemThemeFromWindow` method? That way this code is only written once.

**Review by justindbaur — COMMENTED** [review]
One tiny thing, otherwise looks great!

File: libs/angular/src/platform/services/theming/angular-theming.service.ts (hunk 1 above)
Comment (dani-garcia) [reply libs/angular/src/platform/services/theming/angular-theming.service.ts:39]: Ah, great point! changed it in 0a3d45da02e5a95d1432a441b87241cf0480f30d

**Review by justindbaur — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Comment by alexyangjie** [comment]
Any update on when this pull request will be merged? The memory leak has a huge impact on Safari performance and it is the only reason preventing me from switching to Safari.

**Comment by dani-garcia** [comment]
Hey @alexyangjie, I've got the approval from QA to merge this now, so this should be available in the next minor version release of the extension.

**Pull request merged**